optimum/intel/openvino/modeling_decoder.py

```diff
@@ -249,6 +250,64 @@ class OVModelForCausalLM(OVBaseDecoderModel, GenerationMixin):
     export_feature = "text-generation"
     auto_model_class = AutoModelForCausalLM
 
+    def __init__(
+        self,
+        model: openvino.runtime.Model,
+        config: PretrainedConfig = None,
+        device: str = "CPU",
+        dynamic_shapes: bool = True,
+        ov_config: Optional[Dict[str, str]] = None,
+        model_save_dir: Optional[Union[str, Path, TemporaryDirectory]] = None,
+        **kwargs,
+    ):
+        model = self._try_modify_pastkv_to_lowprecision(model, ov_config, device, **kwargs)
+        super().__init__(model, config, device, dynamic_shapes, ov_config, model_save_dir, **kwargs)
+
+    def _try_modify_pastkv_to_lowprecision(
+        self, model: openvino.runtime.Model, ov_config: Optional[Dict[str, str]] = None, device: str = "CPU", **kwargs
+    ):
+        device = device.upper() if device else device
+        if device == "CPU":
+            pastkv_will_use = core.get_property(device, "INFERENCE_PRECISION_HINT")
+            # ov_config["INFERENCE_PRECISION_HINT"] may override the prefer precision
+            if ov_config:
+                user_hint = ov_config.get("INFERENCE_PRECISION_HINT", "")
+                openvino_types_str_map = {
```

Comment:
is it possible to move type mapping in utils or even on openvino side for avoid duplication if it will requires to get dtype from string or maybe it makes sense to have construction from string for Type for avoid requirements to update mapping if new type will be supported (e.g. i4 for example)

Reply:
Fixed.